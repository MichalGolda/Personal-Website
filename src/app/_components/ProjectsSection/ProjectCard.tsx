import Image from "next/image";
import { Project } from "@/app/_types/project";

export type ProjectCardProps = Project;

const ProjectCard: React.FC<ProjectCardProps> = ({
  name,
  description,
  showUrl,
  githubUrl,
  coverImageSrc,
  ...props
}) => {
  return (
    <div className="flex flex-col gap-y-4" {...props}>
      <h3 className="text-secondary font-bold text-xl md:text-2xl lg:text-3xl">
        {name}
      </h3>
      <div className="flex flex-row gap-x-4">
        {showUrl && (
          <a
            className="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
            href={showUrl}
          >
            Poka≈º
            <Image src="/arrow.png" alt="arrow" width={32} height={0} />
          </a>
        )}
        {githubUrl && (
          <a
            className="text-primary font-bold flex flex-row gap-x-4"
            href={githubUrl}
          >
            Github
            <Image src="/arrow.png" alt="arrow" width={32} height={0} />
          </a>
        )}
      </div>
      <p className="text-body">{description}</p>
      <Image
        className="!relative object-contain !h-fit"
        src={coverImageSrc}
        alt={name}
        fill
      />
    </div>
  );
};

export default ProjectCard;
