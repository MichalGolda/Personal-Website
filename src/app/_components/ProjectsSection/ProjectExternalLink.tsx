import Image from "next/image";

export type ProjectExternalLinkProps = {
  href: string;
  name: string;
};

const ProjectExternalLink = ({ href, name }: ProjectExternalLinkProps) => (
  <a
    className="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
    href={href}
  >
    {name}
    <Image src="/arrow.webp" alt="arrow" width={32} height={0} />
  </a>
);

export default ProjectExternalLink;
