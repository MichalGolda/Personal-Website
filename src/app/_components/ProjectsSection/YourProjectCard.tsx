import { PrimaryButton } from "@/app/_components";

export default function YourProjectCard() {
  return (
    <div className="w-full h-full bg-secondary flex items-center justify-center">
      <div className="flex flex-col gap-y-4 items-center">
        <h4 className="text-white font-bold text-2xl">
          Miejsce na twój projekt
        </h4>
        <a href="#contact">
          <PrimaryButton className="w-fit">Skontaktuj się</PrimaryButton>
        </a>
      </div>
    </div>
  );
}
