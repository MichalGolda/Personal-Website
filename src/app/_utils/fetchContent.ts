import { promises as fs } from "fs";
import path from "path";

export const fetchContent = (contentFileName: string) => 
    fs
        .readFile(`${path.resolve("content")}/${contentFileName}`, "utf-8")
        .then((value) => JSON.parse(value));