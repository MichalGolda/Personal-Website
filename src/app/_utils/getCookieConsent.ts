import { cookies } from "next/headers"; 

export const getCookieConsent = () => {
    const cookiesStore = cookies();
    const cookieConsent = cookiesStore.get("cookieConsent");
  
    return cookieConsent ? JSON.parse(cookieConsent.value) : false;
};