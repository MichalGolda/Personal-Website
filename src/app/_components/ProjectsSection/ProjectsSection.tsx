import { SectionHeading, Container } from "@/app/_components";
import ProjectCard from "./ProjectCard";
import YourProjectCard from "./YourProjectCard";

export default function ProjectsSection() {
  return (
    <div id="portfolio" className="my-64">
      <Container>
        <SectionHeading
          headline="portfolio"
          title="Ostatnio zrezlizowane projekty"
        />
        <div className="grid mt-16 grid-cols-1 auto-cols-fr auto-rows-fr lg:grid-cols-2 lg:grid-rows-2 gap-16">
          <ProjectCard
            title="Pichkrakow.pl"
            description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum ut purus congue, varius neque posuere, egestas magna. Vivamus lacinia urna sed sapien tincidunt, in feugiat libero auctor."
            showUrl="https://pichkrakow.pl/beta"
            coverImageUrl="/pichkrakow.jpg"
          />
          <ProjectCard
            title="Pichkrakow.pl"
            description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum ut purus congue, varius neque posuere, egestas magna. Vivamus lacinia urna sed sapien tincidunt, in feugiat libero auctor."
            showUrl="https://pichkrakow.pl/beta"
            coverImageUrl="/pichkrakow.jpg"
          />
          <ProjectCard
            title="Pichkrakow.pl"
            description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum ut purus congue, varius neque posuere, egestas magna. Vivamus lacinia urna sed sapien tincidunt, in feugiat libero auctor."
            showUrl="https://pichkrakow.pl/beta"
            coverImageUrl="/pichkrakow.jpg"
          />
          <YourProjectCard />
        </div>
      </Container>
    </div>
  );
}
