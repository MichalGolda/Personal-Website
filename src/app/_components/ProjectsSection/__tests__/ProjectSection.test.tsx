import "@/app/_mocks/intersectionObserver.mock";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import ProjectsSection from "../ProjectsSection";
import { fetchProjects } from "@/app/_utils/fetchContent";

describe("ProjectsSection component", () => {
  it("renders unchanged", async () => {
    const data = await fetchProjects();

    const { container } = render(<ProjectsSection data={data} />);

    expect(container).toMatchSnapshot();
  });
});
