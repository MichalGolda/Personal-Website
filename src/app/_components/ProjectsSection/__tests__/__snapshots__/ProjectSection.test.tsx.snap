// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsSection component renders unchanged 1`] = `
<div>
  <div
    class="my-64"
    id="portfolio"
  >
    <div
      class="md:max-w-[1024px] lg:max-w-[1320px] w-full mx-auto md:px-16 px-4 "
    >
      <div
        class="text-center max-w-3xl mx-auto"
      >
        <span
          class="text-primary font-semibold text-md md:text-lg lg:text-xl uppercase"
        >
          portfolio
        </span>
        <h2
          class="text-2xl md:text-3xl lg:text-4xl text-secondary font-bold"
        >
          Ostatnio zrezlizowane projekty
        </h2>
      </div>
      <div
        class="grid mt-16 grid-cols-1 auto-cols-fr auto-rows-fr lg:grid-cols-2 lg:grid-rows-2 gap-16"
      >
        <div
          class="flex flex-col gap-y-4"
          data-aos="zoom-in"
          data-aos-duration="500"
        >
          <h3
            class="text-secondary font-bold text-xl md:text-2xl lg:text-3xl"
          >
            Pichkrakow.pl
          </h3>
          <div
            class="flex flex-row gap-x-4"
          >
            <a
              class="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
              href="https://pichkrakow.pl/beta"
            >
              Podgląd
              <img
                alt="arrow"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/_next/image?url=%2Farrow.webp&w=64&q=75"
                srcset="/_next/image?url=%2Farrow.webp&w=32&q=75 1x, /_next/image?url=%2Farrow.webp&w=64&q=75 2x"
                style="color: transparent;"
                width="32"
              />
            </a>
          </div>
          <p
            class="text-body"
          >
            Strona internetowa dla Gabinetu Psychoterapeutycznego
          </p>
          <img
            alt="Pichkrakow.pl"
            class="!relative object-contain !h-fit"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fpichkrakow.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fpichkrakow.webp&w=640&q=75 640w, /_next/image?url=%2Fpichkrakow.webp&w=750&q=75 750w, /_next/image?url=%2Fpichkrakow.webp&w=828&q=75 828w, /_next/image?url=%2Fpichkrakow.webp&w=1080&q=75 1080w, /_next/image?url=%2Fpichkrakow.webp&w=1200&q=75 1200w, /_next/image?url=%2Fpichkrakow.webp&w=1920&q=75 1920w, /_next/image?url=%2Fpichkrakow.webp&w=2048&q=75 2048w, /_next/image?url=%2Fpichkrakow.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="flex flex-col gap-y-4"
          data-aos="zoom-in"
          data-aos-duration="500"
        >
          <h3
            class="text-secondary font-bold text-xl md:text-2xl lg:text-3xl"
          >
            Animal-Houses
          </h3>
          <div
            class="flex flex-row gap-x-4"
          >
            <a
              class="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
              href="https://animal-houses.vercel.app/"
            >
              Podgląd
              <img
                alt="arrow"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/_next/image?url=%2Farrow.webp&w=64&q=75"
                srcset="/_next/image?url=%2Farrow.webp&w=32&q=75 1x, /_next/image?url=%2Farrow.webp&w=64&q=75 2x"
                style="color: transparent;"
                width="32"
              />
            </a>
            <a
              class="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
              href="https://github.com/michalgolda/animal-houses"
            >
              Github
              <img
                alt="arrow"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/_next/image?url=%2Farrow.webp&w=64&q=75"
                srcset="/_next/image?url=%2Farrow.webp&w=32&q=75 1x, /_next/image?url=%2Farrow.webp&w=64&q=75 2x"
                style="color: transparent;"
                width="32"
              />
            </a>
          </div>
          <p
            class="text-body"
          >
            Aplikacja do zarządzania sprzedażą domków dla zwierząt.
          </p>
          <img
            alt="Animal-Houses"
            class="!relative object-contain !h-fit"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fanimal-houses.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fanimal-houses.webp&w=640&q=75 640w, /_next/image?url=%2Fanimal-houses.webp&w=750&q=75 750w, /_next/image?url=%2Fanimal-houses.webp&w=828&q=75 828w, /_next/image?url=%2Fanimal-houses.webp&w=1080&q=75 1080w, /_next/image?url=%2Fanimal-houses.webp&w=1200&q=75 1200w, /_next/image?url=%2Fanimal-houses.webp&w=1920&q=75 1920w, /_next/image?url=%2Fanimal-houses.webp&w=2048&q=75 2048w, /_next/image?url=%2Fanimal-houses.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="flex flex-col gap-y-4"
          data-aos="zoom-in"
          data-aos-duration="500"
        >
          <h3
            class="text-secondary font-bold text-xl md:text-2xl lg:text-3xl"
          >
            Cashly
          </h3>
          <div
            class="flex flex-row gap-x-4"
          >
            <a
              class="text-primary text-sm md:text-base font-bold flex flex-row gap-x-4"
              href="https://github.com/michalgolda/cashly"
            >
              Github
              <img
                alt="arrow"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/_next/image?url=%2Farrow.webp&w=64&q=75"
                srcset="/_next/image?url=%2Farrow.webp&w=32&q=75 1x, /_next/image?url=%2Farrow.webp&w=64&q=75 2x"
                style="color: transparent;"
                width="32"
              />
            </a>
          </div>
          <p
            class="text-body"
          >
            Aplikacja do zarządzania prywatnymi finansami.
          </p>
          <img
            alt="Cashly"
            class="!relative object-contain !h-fit"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fcashly.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fcashly.webp&w=640&q=75 640w, /_next/image?url=%2Fcashly.webp&w=750&q=75 750w, /_next/image?url=%2Fcashly.webp&w=828&q=75 828w, /_next/image?url=%2Fcashly.webp&w=1080&q=75 1080w, /_next/image?url=%2Fcashly.webp&w=1200&q=75 1200w, /_next/image?url=%2Fcashly.webp&w=1920&q=75 1920w, /_next/image?url=%2Fcashly.webp&w=2048&q=75 2048w, /_next/image?url=%2Fcashly.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="w-full h-full bg-secondary flex items-center justify-center"
          data-aos="zoom-in"
          data-aos-duration="500"
        >
          <div
            class="flex flex-col gap-y-4 items-center"
          >
            <h4
              class="text-white text-lg md:text-xl lg:text-2xl font-bold"
            >
              Miejsce na twój projekt
            </h4>
            <a
              href="#contact"
            >
              <button
                class="py-4 px-8 font-bold text-sm md:text-base transition-all duration-500 disabled:bg-lightGrey disabled:text-body bg-primary text-white hover:bg-primary/90 active:bg-primary/80 border border-transparent border-solid w-fit"
              >
                Skontaktuj się
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
